declarations_site/chatbot/utils.py

```diff
@@ -35,10 +43,81 @@ def simple_search(query, deepsearch=False):
     return search
 
 
+def botframework_jwt_keys():
+    # This is a static URL that you can hardcode into your application.
+    openidURL = 'https://login.botframework.com/v1/.well-known/openidconfiguration'
+    cached_keys = cache.get(openidURL)
+    if cached_keys:
+        return cached_keys
+
+    response = requests.get(openidURL, timeout=10)
+    config = response.json()
+
+    if config['issuer'] != 'https://api.botframework.com':
+        return False
+
+    response = requests.get(config['jwks_uri'], timeout=10)
+    keys = response.json()
+    if 'keys' in keys and keys['keys']:
+        cache.set(openidURL, keys, 86400)
```

Comment:
I do not see space for fine tuning here
In MS library these values are hardcoded